packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1643,7 +1643,11 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       }
     }
 
-    _formatAndSetValue(value);
+    if (_isSelectionOnlyChange(value)) {
```

Comment:
Does the web go through the text input client instead of raw keyboard event?

Reply:
Yes. All keyboard interactions with the text field are handled by the browser and we only receive the result of the change. e.g. If the user clicks "shift+right arrow", we receive an event from the browser telling us that selection has changed, so we read the new selection state from the input field and we send it to Flutter through the text input client.